src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -2103,6 +2103,15 @@ private function registerCacheConfiguration(array $config, ContainerBuilder $con
                 'tags' => false,
             ];
         }
+
+        if ('cache.adapter.redis_tag_aware' === $config['app']) {
```

Comment:
In theory, any pool can inherit from redis_tag_aware and require making cache.app an alias

Reply:
If I understand you correctly, the same could be said for the non-app cache pools below, correct (`cache.adapter.redis_tag_aware` is hard-coded)?

Reply:
absolutely, recursively also, since pools can derivate from other pools.

Reply:
Ultimately, I don't think there should be any redis-specific check here at all. As discussed in slack, we should be using a pass to determine if the adapter used is already tag aware and alias/decorate based on that.

What do you think about matching what we do for custom pools now for 5.3 as a bug fix, then adding a more robust system in 6.1?

Reply:
but the alias is created in the DI extension - the pass doesn't know about them

Reply:
Think I finally understand and I believe #44682 solves.